### PR TITLE
WooCommerce Orders: Add state for creating notes

### PR DESCRIPTION
**Pull request by ryelle**
This is infrastructure-only. This PR adds a reducer, action, and selector which will be used for creating notes on orders. There is no UI yet.

To test:

- Run the tests: `npm test` or `npm run test-client client/extensions/woocommerce/state/sites/orders/notes/`

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/orders/notes/reducer.js [hunk 1]
```diff
@@ -34,6 +37,26 @@ export function isLoading( state = {}, action ) {
...
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function isSaving( state = {}, action ) {
+	switch ( action.type ) {
+		case WOOCOMMERCE_ORDER_NOTE_CREATE:
+		case WOOCOMMERCE_ORDER_NOTE_CREATE_FAILURE:
+		case WOOCOMMERCE_ORDER_NOTE_CREATE_SUCCESS:
+			return Object.assign( {}, state, { [ action.orderId ]: WOOCOMMERCE_ORDER_NOTE_CREATE === action.type } );
```
Comment (belcherj) [inline client/extensions/woocommerce/state/sites/orders/notes/reducer.js]: Why not use the spread operator?

File: client/extensions/woocommerce/state/sites/orders/notes/selectors.js [hunk 2]
```diff
@@ -50,3 +50,15 @@ export const getOrderNotes = ( state, orderId, siteId = getSelectedSiteId( state
...
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {Number} orderId Order ID to check.
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {boolean} Whether we're currently saving a note for a given order on a site.
+ */
+export const isOrderNoteSaving = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	const isSaving = get( state, [ 'extensions', 'woocommerce', 'sites', siteId, 'orders', 'notes', 'isSaving', orderId ] );
+	// Strict check because it could also be undefined.
+	return ( true === isSaving );
```
Comment (belcherj) [inline client/extensions/woocommerce/state/sites/orders/notes/selectors.js]: do you need `true ===`, does isSaving return truthy values when not true?

File: client/extensions/woocommerce/state/sites/orders/notes/reducer.js (hunk 1 above)
Comment (ryelle) [inline client/extensions/woocommerce/state/sites/orders/notes/reducer.js]: Habit :) and isLoading (in this file) uses Object.assign (along with the other reducers I've written). If we have standardized on the spread operator instead I can make another PR to change those next week.

File: client/extensions/woocommerce/state/sites/orders/notes/selectors.js (hunk 2 above)
Comment (ryelle) [inline client/extensions/woocommerce/state/sites/orders/notes/selectors.js]: No, I just wanted to ensure that a boolean is returned.

**Pull request merged**